src/shared/install.h

```diff
@@ -95,88 +95,88 @@ struct UnitFileInstallInfo {
         bool auxiliary;
 };
 
-int unit_file_enable(
+int install_unit_enable(
```

Comment:
I am not convinced this renaming is really a good idea. The D-Bus calls PID 1 exposes are all called UnitFileThis and ThatUnitFile, and that's API.

I think there's more value in keeping conistency between the dbus apis and the underlying C apis, than bikeshedding about the names...

Reply:
The issue I was trying to solve is that the naming in the code is just misleading. E.g. you look at the code, and it says "files", but that's actually "unit names". And "unmask file", but it's really operating on a unit, which might have multiple files, etc. This creates a cognitive overhead in code which is very complicated anyway.

The D-Bus API cannot be changed, but the D-Bus API is something that is relatively simple and unchanging, so the amount of time spent looking at install.c is much much larger than the time spent looking at the D-bus wrappers which are just pass-through for the most part.

So I'd prefer to do the renaming, and maybe even follow it up with other renamings in the future.

Reply:
i don't follow? this code this deals with files on disk, not with the runtime objects that these files create. hence unit files is exactly what this is about? I mean, sure, sometimes it's symlinks too, but that's effect how unix uses the word "files" for all types inodes usually.

Reply:
I'll paste the text from the commit message:

> The original names were conceived when a unit was equivalent to a unit file.
But the meaning has shifted over time, and while it is meanigful to say
"unit_file_enable", it is less so to say "unit_file_disable", since the
disablement can happen without a unit file being present. And we're not really
disabling the *file*, because the file is unchanged, and it's just the
surrounding state of that changes.

> Those functions are defined in install.[ch], so let's call them install_unit_*.
(This should be read as "install_(unit_enable)", not "(install_unit)_enable".)
And UnitFileChange → InstallChange.

Reply:
Hmm? the symlinks to the unit file are removed. They string the symlinks point to reference the unit file, if it is missing or not. You cannot enable a unit really, only unit files, since the symlinks must point to something, which is a unit file, regardless if absent or not.

Reply:
i mean if the symlinks wouldn't point to unit file paths i could follow your thinking, but they *do* point to unit file paths

Reply:
Let's start from other side. `unit_file_disable` does not operate on unit files. It operates on symlinks, which might point to the unit file, but also to other locations. And for the user, we say "disable unit", "could not disable unit", etc. This is particularly important in case of aliases, where we may operate on a single enablement symlink. If we were starting from scratch today, we certainly wouldn't have "file" in the name there. As I wrote above, this renaming is more relevant for some cases, less important for others. But it'd be ugly and confusing to have e.g. `install_unit_disable` and `unit_file_enable`, so I renamed them all to follow a common pattern.

Reply:
> but also to other locations

Like where?

> Let's start from other side. unit_file_disable does not operate on unit files. It operates on symlinks, which might point to the unit file, but also to other locations.

I still don't follow. For me "enabling" means something like "hooking in", "linking up", "setting up for activation", "empowering for doing", but all this words don't actually describe an operation on the think you enable, but on axuiliary stuff oyu have to do for the thing to be enabled... So I am *really* not getting your point, at all.

I mean, I am not a native speaker. But I honestly, and truely don't follow what you are trying to say.

If I enable a car to got from A to B by building a road, then I build that road, and don't actually modify the car... If I fill up the car with petrol, then I move the petrol, I don't really modify the car. It's always the other thing you operate on to allow the thing you enable to do or be something.

Same here: to enable a unit file we create symlinks. But it's eally the unit file we enable, not the unit. We can't enable units at all.

Anyway, puzzled by this